libraries/chain/webassembly/runtimes/eos-vm-oc/executor.cpp

```diff
@@ -281,6 +284,12 @@ void executor::execute(const code_descriptor& code, memory& mem, host_context& c
       //case 1: clean eosio_exit
       case EOSVMOC_EXIT_CHECKTIME_FAIL:
          context.trx_context.checktime();
+
+         // If EOSVMOC_EXIT_CHECKTIME_FAIL is caught, context.trx_context.checktime()
+         // must throw. Otherwise we would have interrupted contract execution
+         // at some unknown time but still considered it completed successfully.
+         assert(false);
+         __builtin_unreachable();
```

Comment:
btw if we don't want to put this noise here (and below for eos-vm too), maybe we could make a `[[noreturn]] checktime_must_throw()` to be used in these two locations, and move this check inside of `checktime_must_throw()`. Not required; just thinking about a way to deduplicate.,

Reply:
https://github.com/AntelopeIO/spring/pull/1525/commits/7db87ec5a9cc5a1af9fb4943c9f53b0e9c0c61a5